File: api-tester/src/defaults/java/com/revenuecat/apitester/java/RedeemRCBillingPurchaseListenerAPI.java
package com.revenuecat.apitester.java;

import androidx.annotation.OptIn;

import com.revenuecat.purchases.ExperimentalPreviewRevenueCatPurchasesAPI;
import com.revenuecat.purchases.interfaces.RedeemRCBillingPurchaseListener;

import kotlin.Unit;
import kotlin.jvm.functions.Function1;

@OptIn(markerClass = ExperimentalPreviewRevenueCatPurchasesAPI.class)
@SuppressWarnings({"unused"})
final class RedeemRCBillingPurchaseListenerAPI {
    static void checkListener(RedeemRCBillingPurchaseListener listener,
                              RedeemRCBillingPurchaseListener.RedemptionStarter redemptionStarter) {
        listener.handleRCBillingPurchaseRedemption(redemptionStarter);
    }

    static void checkRedemptionStarter(RedeemRCBillingPurchaseListener.RedemptionStarter redemptionStarter,
                                       RedeemRCBillingPurchaseListener.ResultListener resultListener) {
        redemptionStarter.startRedemption(resultListener);
    }

    static void checkResultListener(RedeemRCBillingPurchaseListener.ResultListener resultListener,
                                    RedeemRCBillingPurchaseListener.RedeemResult result) {
        resultListener.handleResult(result);
    }

    static void checkRedeemResult(RedeemRCBillingPurchaseListener.RedeemResult result) {
        switch (result) {
            case SUCCESS:
            case ERROR:
                break;
        }
    }
}

File: api-tester/src/defaults/kotlin/com/revenuecat/apitester/kotlin/RedeemRCBillingPurchaseListenerAPI.kt
package com.revenuecat.apitester.kotlin

import com.revenuecat.purchases.ExperimentalPreviewRevenueCatPurchasesAPI
import com.revenuecat.purchases.Purchases
import com.revenuecat.purchases.interfaces.RedeemRCBillingPurchaseListener

@OptIn(ExperimentalPreviewRevenueCatPurchasesAPI::class)
@Suppress("unused", "UNUSED_VARIABLE")
private class RedeemRCBillingPurchaseListenerAPI {
    fun checkListener(
        redeemRCBillingPurchaseListener: RedeemRCBillingPurchaseListener,
        redemptionStarter: RedeemRCBillingPurchaseListener.RedemptionStarter,
    ) {
        redeemRCBillingPurchaseListener.handleRCBillingPurchaseRedemption(redemptionStarter)
    }

    fun checkRedemptionStarter(
        redemptionStarter: RedeemRCBillingPurchaseListener.RedemptionStarter,
        resultListener: RedeemRCBillingPurchaseListener.ResultListener,
    ) {
        redemptionStarter.startRedemption(resultListener)
    }

    fun checkResultListener(
        resultListener: RedeemRCBillingPurchaseListener.ResultListener,
        result: RedeemRCBillingPurchaseListener.RedeemResult,
    ) {
        resultListener.handleResult(result)
    }

    fun checkResult(result: RedeemRCBillingPurchaseListener.RedeemResult): Boolean {
        when (result) {
            RedeemRCBillingPurchaseListener.RedeemResult.SUCCESS,
            RedeemRCBillingPurchaseListener.RedeemResult.ERROR,
            -> {
                return true
            }
        }
    }

    fun checkPurchases(
        purchases: Purchases,
        redeemRCBillingPurchaseListener: RedeemRCBillingPurchaseListener,
    ) {
        purchases.redeemRCBillingPurchaseListener = redeemRCBillingPurchaseListener
        val getListener: RedeemRCBillingPurchaseListener? = purchases.redeemRCBillingPurchaseListener
    }
}
